src/vs/workbench/contrib/notebook/browser/viewParts/notebookEditorStickyScroll.ts

```diff
@@ -0,0 +1,342 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as DOM from 'vs/base/browser/dom';
+import { Disposable, DisposableStore } from 'vs/base/common/lifecycle';
+import { INotebookEditor } from 'vs/workbench/contrib/notebook/browser/notebookBrowser';
+import { NotebookOptions } from 'vs/workbench/contrib/notebook/browser/notebookOptions';
+import { INotebookCellList } from 'vs/workbench/contrib/notebook/browser/view/notebookRenderingCommon';
+import { NotebookCellOutlineProvider, OutlineEntry } from 'vs/workbench/contrib/notebook/browser/viewModel/notebookOutlineProvider';
+import { CellKind } from 'vs/workbench/contrib/notebook/common/notebookCommon';
+
+
+export class NotebookEditorStickyScroll extends Disposable {
+	private _stickyScrollContent: StickyScrollContent | undefined;
+	private readonly _disposables = new DisposableStore();
+
+	constructor(
+		private readonly domNode: HTMLElement,
+		private readonly notebookEditor: INotebookEditor,
+		private readonly notebookOptions: NotebookOptions,
+		private readonly notebookOutline: NotebookCellOutlineProvider,
+		private readonly notebookCellList: INotebookCellList
+	) {
+		super();
+
+		this._stickyScrollContent = new StickyScrollContent(this.notebookEditor, this.notebookOutline, this.notebookCellList, this.notebookOptions);
```

Comment:
`StickyScrollContent` needs to be added to the disposable store (`this._register`)